pkg/util/kubernetes/apiserver/store.go

```diff
@@ -55,27 +55,24 @@ func (m *metaBundleStore) get(nodeName string) (*metadataMapperBundle, bool) {
 	return metaBundle, true
 }
 
-func (m *metaBundleStore) getOrCreate(nodeName string) *metadataMapperBundle {
+func (m *metaBundleStore) getCopyOrNew(nodeName string) *metadataMapperBundle {
 	cacheKey := agentcache.BuildAgentKey(metadataMapperCachePrefix, nodeName)
 
-	var metaBundle *metadataMapperBundle
+	metaBundle := newMetadataMapperBundle()
 
 	m.mu.Lock()
 	defer m.mu.Unlock()
 
 	v, ok := m.cache.Get(cacheKey)
```

Comment:
where are we setting the cache?

Reply:
we set in the cache outside this method. I change the name in order to reflect the change.
I also check every where this method was call, in order to be sure that it works as before